### PR TITLE
Prompt user to update if API version check fails

### PR DESCRIPTION
**Pull request by beneills**
If there's a serious server error, we display:

"We encountered an unknown error while querying the Genestack API.  Please verify that you're running the latest version via: `pip install https://github.com/genestack/python-client/archive/stable.zip --upgrade`"

### CODE REVIEW

File: genestack_client/genestack_exceptions.py [hunk 1]
```diff
@@ -28,12 +28,13 @@ def __init__(self, message, path, post_data, debug=False, stack_trace=None):
...
         message = message.encode('utf-8', 'ignore') if isinstance(message, unicode) else message
 
-        GenestackException.__init__(self, message, path, post_data, debug, stack_trace)
+        GenestackException.__init__(self, message, path, post_data, debug, stack_trace, prompt_update)
         self.message = message
         self.debug = debug
         self.stack_trace = stack_trace
         self.path = path
         self.post_data = post_data
+        self.prompt_update = prompt_update
```
Comment (vasiliy-bout) [inline genestack_client/genestack_exceptions.py]: This flag looks like a dirty hack. We must not merge this branch.

File: genestack_client/genestack_connection.py [hunk 2]
```diff
@@ -95,8 +95,14 @@ def check_version(self):
...
-        compatible_version = self.application('genestack/clientVersion'
-                                             ).invoke('getCompatibleVersion')
+        try:
+            compatible_version = self.application('genestack/clientVersion'
+                                                 ).invoke('getCompatibleVersion')
+        except GenestackServerException, e:
+            # We don't know what happened, but it might be due to incompatible client/API versions.
+            # Rethrow exception, making sure we tell the user to update.
+            e.prompt_update = True
+            raise e
```
Comment (vasiliy-bout) [inline genestack_client/genestack_connection.py]: We can simply throw here a `GenestackVersionException` with corresponding message.

Currently message is hardcoded inside `GenestackVersionException`. We can change this logic and provide the message to constructor.

**Comment by genestack-solomatin** [comment]
And there is one more idea, replace exception throw with.
```
sys.stderr.write(message)
exit(13)  
```

File: genestack_client/genestack_connection.py (hunk 2 above)
Comment (beneills) [inline genestack_client/genestack_connection.py]: Fixed.

File: genestack_client/genestack_exceptions.py (hunk 1 above)
Comment (beneills) [inline genestack_client/genestack_exceptions.py]: Fixed.

**Pull request merged**